app.rb

```diff
@@ -16,84 +16,88 @@ def clear_popolo
   end
 end
 
-class GBCountryStatsGenerator
-  include Sidekiq::Worker
-  include Everypoliticianbot::Github
-
-  def perform(countries_json_url)
-    Everypolitician.countries_json = countries_json_url
-    with_git_repo('everypolitician/gender-balance-country-stats', branch: 'gh-pages', message: 'Update stats.json') do
-
-      gender_stats = []
-      EveryPolitician.countries.each do |c|
-        country = {
-          slug: c.slug,
+class GenderStats
+  def self.report
+    gender_stats = []
+    EveryPolitician.countries.take(1).each do |c|
+      country = {
+        slug: c.slug,
+      }
+      legislatures = []
+      c.legislatures.each do |l|
+        totals_seen = {}
+        totals = {
+          overall: Hash.new(0),
+          parties: Hash.new { | hash, key | hash[key] = Hash.new(0) },
+        }
+        legislature = {
+          slug: l.slug,
         }
-        legislatures = []
-        c.legislatures.each do |l|
-          totals_seen = {}
-          totals = {
+        terms = {}
+
+        l.popolo.events.where(classification: "legislative period").each do |lp|
+          terms[lp.id] = {
             overall: Hash.new(0),
             parties: Hash.new { | hash, key | hash[key] = Hash.new(0) },
           }
-          legislature = {
-            slug: l.slug,
-          }
-          terms = {}
+        end
 
-          l.popolo.events.where(classification: "legislative period").each do |lp|
-            terms[lp.id] = {
-              overall: Hash.new(0),
-              parties: Hash.new { | hash, key | hash[key] = Hash.new(0) },
-            }
-          end
+        person_memberships = Hash.new { | hash, key | hash[key] = [] }
+        l.popolo.memberships.each do |m|
+          person_memberships[m.person_id].push(m)
+        end
```

Comment:
This is quite hard to follow. Is this really just a hand-rolled version of `person_memberships = l.popolo.memberships.group_by(&:person_id)`?